File: src/components/replicationGroup.ts
import { t } from "@rbxts/t";
import { defineComponent } from "../../src";

const definition = defineComponent({
	name: "ReplicationGroup",
	data: <{ [key in keyof Components]?: string }>(<unknown>true),
	refValidator: t.any,
});

declare global {
	interface Components {
		/** The ReplicationGroup Component holds the ReplicationGroups that other Components on this Entity should fall under
		 *
		 * Data is a Map of Component name to ReplicationGroupId
		 */
		ReplicationGroup: typeof definition;
	}
}

export = definition;

File: src/components/replicationGroup.lua
local TS = _G[script]
local t = TS.import(script, TS.getModule(script, "@rbxts", "t").lib.ts).t
local defineComponent = TS.import(script, script.Parent.Parent, "defineComponent").default
local definition = defineComponent({
	name = "ReplicationGroup",
	data = true,
	refValidator = t.any,
})
return definition
